APISpec.md

```diff
@@ -0,0 +1,508 @@
+# Todo API Spec
+
+## API Resources
+
+* **User** - A User represents an owner of Todos and corresponds one-to-many to a Todo.
+
+* **Todo** - A Todo represents a task to be completed and corresponds many-to-one to a User.
+
+* **Session** - A Session represents an authenticated user session and corresponds many-to-one to a User.
+
+## Data Tables
+
+### Users:
+```
+ID - Guid (PK)
+Username - String
+Password - String (Hash)
+```
+
+### Todos:
+```
+ID - Guid (PK)
+UserID - Guid (FK - ID in Users)
+Title - String
+Status - Enumified String
+CreationDate - Datetime
+DueDate - Datetime
+```
+
+### Tokens:
+```
+ID - Guid (PK)
+UserID - Guid (FK - ID in Users)
+Token - String
+Expiration - Datetime
+```
+
+## Data Models
+
+### User
+```
+ID - Guid
+Username - String
+Password - String (Hash)
+Todos - Todo[] (List of Todos for the user)
+Sessions - Session[] (List of active sessions for the user)
```

Comment:
I would say very rarely you would need this functionality and only if you have an auditing system to be honest.

Reply:
Yeah, that makes sense